### PR TITLE
Use amazon linux 2023 runners for Docker builds

### PR DESCRIPTION
**Pull request by atalman**
Migrate these builds to linux 2023. We want to build and test the Docker images in CD.

Looks like we are hitting this issue: https://github.com/docker/buildx/issues/379 when trying to build Docker on Amazon Linux 2023.

Conda Docker build is timing out. While Manywheel is executing but failing because BUILDKIT is turned off: https://github.com/pytorch/pytorch/actions/runs/11036043157/job/30653543264?pr=136544

Proposed Solution is to fix it in user_data . Please see: https://github.com/pytorch/test-infra/issues/5712

I see docker builds are executed successfully here: https://github.com/pytorch/pytorch/actions/runs/11040149229/job/30667448668?pr=136544

Workaround timeout problem (reported in https://bugzilla.redhat.com/show_bug.cgi?id=1537564 ) by configuring number of open files per container to 1048576

### CODE REVIEW

**Review by ZainRizvi — DISMISSED** [review]

File: .ci/docker/manywheel/build.sh [hunk 1]
```diff
@@ -124,7 +124,12 @@ if [[ -n ${MANY_LINUX_VERSION} && -z ${DOCKERFILE_SUFFIX} ]]; then
 fi
 (
     set -x
-    DOCKER_BUILDKIT=1 docker build \
+    containerd --version
+    cat /usr/lib/systemd/system/containerd.service | grep LimitNOFILE
+    echo -e '[Service]\nLimitNOFILE=1048576' > /etc/systemd/system/docker.service.d/ulimit.conf
```
Comment (malfet) [inline .ci/docker/manywheel/build.sh:129]:
```suggestion
    grep LimitNOFILE  /usr/lib/systemd/system/docker.service
    sudo sed -i s/LimitNOFILE=infinity/LimitNOFILE=1048576/ /usr/lib/systemd/system/docker.service
```

**Review by ZainRizvi — COMMENTED** [review]
Why not resolve the fix in test-infra first and then merge the pytorch PR after it starts consuming those test-infra changes? That would help test the changes in a clean environment

File: .ci/docker/conda/Dockerfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ FROM centos:7 as base
 ENV LC_ALL en_US.UTF-8
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US.UTF-8
+RUN echo "ulimit is $(ulimit -n)"
```
Comment (malfet) [inline .ci/docker/conda/Dockerfile:8]:
```suggestion
```

File: .ci/docker/manywheel/Dockerfile [hunk 3]
```diff
@@ -10,6 +10,8 @@ ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US.UTF-8
 
 ARG DEVTOOLSET_VERSION=9
+RUN echo "ulimit is $(ulimit -n)"
```
Comment (malfet) [inline .ci/docker/manywheel/Dockerfile:13]:
```suggestion
```

**Review by ZainRizvi — APPROVED** [review]

File: .ci/docker/conda/build.sh [hunk 4]
```diff
@@ -37,6 +37,12 @@ esac
 
 (
   set -x
+  # TODO: Remove LimitNOFILE=1048576 patch once https://github.com/pytorch/test-infra/issues/5712
+  # is resolved. This patch is required in order to fix timing out of Docker build on Amazon Linux 2023.
+  sudo sed -i s/LimitNOFILE=infinity/LimitNOFILE=1048576/ /usr/lib/systemd/system/docker.service
```
Comment (ZainRizvi) [inline .ci/docker/conda/build.sh:42]: nit: please add a link to the issue that you discovered which explains the problem in more detail

**Review by ZainRizvi — APPROVED** [review]

**Pull request closed without merging**